manifests/vhost.pp

```diff
@@ -618,11 +618,9 @@
 
   # Template uses:
   # - $additional_includes
-  if $additional_includes and ! empty($additional_includes) {
-    concat::fragment { "${name}-additional_includes":
-      target  => "apache::vhost::${name}",
-      order   => 70,
-      content => template('apache/vhost/_additional_includes.erb'),
+  if $additional_includes and ! empty($additional_includes) and $ensure == 'present' {
```

Comment:
A lot of these resources only need to be included if ```$ensure == 'present'```. Maybe it should be a big if statement around a lot of resources rather than duplicated everywhere.